src/osd/ECBackend.cc

```diff
@@ -2295,11 +2323,51 @@ void ECBackend::objects_read_and_reconstruct(
   }
 
   map<hobject_t, set<int>> obj_want_to_read;
-  set<int> want_to_read;
-  get_want_to_read_shards(&want_to_read);
```

Comment:
this seems like a much larger change than necessary... here's all it took to make recovery read the minimum data required: https://github.com/ceph/ceph/pull/21273/commits/468ad4b41010488c8d48ef65ccbebfdb4270690f

Reply:
This modification only relates the read from client. The read of recovery is left as it is. So, don't worry about [468ad4b](https://github.com/ceph/ceph/commit/468ad4b41010488c8d48ef65ccbebfdb4270690f)

Reply:
I'm suggesting that the client read path can be similarly simple as the recovery read path. There should not be redundancies with the EC plugin or the recovery code.